R/check.R

```diff
@@ -1,90 +1,186 @@
+#' Allows string vectors of length 1
+check_single_string_param <- function(name, value, required = TRUE) {
+  if (is.null(value)) {
+    if (required) {
+      rlang::abort(paste0("argument ", name, " is not a single string"),
+        name = name, value = value, class = "invalid_argument"
+      )
+    }
+  } else if (is.character(value)) {
+    if (length(value) != 1) {
+      rlang::abort(paste0("argument ", name, " is not a single string"),
+        name = name, value = value, class = "invalid_argument"
+      )
+    }
+  } else {
+    rlang::abort(paste0("argument ", name, " is not a single string"),
+      name = name, value = value, class = "invalid_argument"
+    )
+  }
+}
+
+#' Allows string vectors of length > 1 and lists of string vectors of length 1
 check_string_param <- function(name, value, required = TRUE) {
-  # string or string[]
-  if ((required &&
-    !(!is.null(value) &&
-      is.character(value))) ||
-    (!required && !(is.null(value) || is.character(value)))) {
+  if (is.null(value)) {
+    if (required) {
+      rlang::abort(paste0("argument ", name, " is not a string"),
+        name = name, value = value, class = "invalid_argument"
+      )
+    }
+  } else if (is.character(value)) {
+    return()
+  } else if (is.list(value)) {
+    for (i in seq_along(value)) {
+      if (!is.character(value[[i]]) || length(value[[i]]) != 1) {
+        rlang::abort(paste0("argument ", name, "[", i, "] is not a single string"),
+          name = name, value = value, class = "invalid_argument"
+        )
+      }
+    }
+  } else {
     rlang::abort(paste0("argument ", name, " is not a string"),
       name = name, value = value, class = "invalid_argument"
     )
   }
 }
 
+#' Allows a single numeric value
+check_single_int_param <- function(name, value, required = TRUE) {
+  if (is.null(value)) {
+    if (required) {
+      rlang::abort(paste0("argument ", name, " is not a single number"),
+        name = name, value = value, class = "invalid_argument"
+      )
+    }
+  } else if (is.numeric(value)) {
```

Comment:
issue: this isn't a check that it's an integer.  If we already use rlang then I'd suggest `rlang::is_integerish` and also call this check `integerish`.  If we really mean integer-values-specifically-with-integer-type, then `is.integer` rather than `is.numeric`.  + update "number" below to describe whatever notion of integer is settled on.
OR: rename `int` checks to, e.g., `numeric`.  If you want to use `checkmate`'s terminology, a single numeric would be a `number`.

Reply:
I changed these checks to `is.integer` and found that it returned FALSE for `value = 1`, which I could see being annoying. So I stuck with `is.numeric`, but I agree, it's too flexible.

Integerish seems perfect for this, so I'll change to that, thanks.